src/Console/Command/Internal/InternalCreateDataSetCommand.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Acquia\Orca\Console\Command\Internal;
+
+use Acquia\Orca\Console\Helper\StatusCode;
+use Acquia\Orca\Helper\Log\TelemetryClient;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Provides a command.
+ */
+class InternalCreateDataSetCommand extends Command {
+
+  /**
+   * The default command name.
+   *
+   * @var string
+   */
+  public static $defaultName = 'internal:create-dataset';
+
+  /**
+   * The telemetry client.
+   *
+   * @var \Acquia\Orca\Helper\Log\TelemetryClient
+   */
+  private $telemetryClient;
+
+  /**
+   * Constructs an instance.
+   *
+   * @param \Acquia\Orca\Helper\Log\TelemetryClient $telemetry_client
+   *   The telemetry client.
+   */
+  public function __construct(TelemetryClient $telemetry_client) {
+    $this->telemetryClient = $telemetry_client;
+
+    parent::__construct(self::$defaultName);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    $this
+      ->setDescription('Creates/updates a domo data stream based on current telemetry config yml')
+      ->addOption('simulate', ['s'], InputOption::VALUE_NONE, 'Run in simulated mode: show what would be created instead of actually creating the stream')
+      ->setHidden(TRUE);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function execute(InputInterface $input, OutputInterface $output): int {
+    $simulate = $input->getOption('simulate');
+
+    // Get the Telemetry Config
+    $data = $this->telemetryClient->getLocalConfig();
```

Comment:
Command classes handle CLI interaction only (i.e., user input and exit codes). Detailed logic is delegated to domain objects, like `\Acquia\Orca\Domain\Fixture\FixtureCreator` or `\Acquia\Orca\Domain\Fixture\FixtureInspector`. That makes testing both a lot easier. (Existing command classes currently have almost 100% test coverage.) Input that requires validation is done via options classes like `\Acquia\Orca\Domain\Fixture\FixtureOptions`. (Those currently have _exactly_ 100% test coverage.)